src/collections/match/operations/match-transfer-item.operation.ts

```diff
@@ -0,0 +1,210 @@
+  ) {
+    this.matchStorage =
+      matchStorage ??
+      new BlDocumentStorage(BlCollectionName.Matches, matchSchema);
+    this.userDetailStorage =
+      userDetailStorage ??
+      new BlDocumentStorage(BlCollectionName.UserDetails, userDetailSchema);
+  }
+
+  private isValidBlid(scannedText: string): boolean {
+    if (Number.isNaN(Number(scannedText))) {
+      if (scannedText.length === 12) {
+        return true;
+      }
+    } else if (scannedText.length === 8) {
+      return true;
+    }
+    return false;
+  }
+
+  async run(blApiRequest: BlApiRequest): Promise<BlapiResponse> {
+    let userFeedback;
+    const blid = blApiRequest?.data?.blid;
+    if (!blid || typeof blid !== "string" || blid.length === 0) {
+      throw new BlError(
+        "blid must be a string with length greater than 0"
+      ).code(803);
+    } else if (!this.isValidBlid(blid)) {
+      throw new BlError("blid is not a valid blid").code(803);
+    }
+
+    const receiverUserDetailId = blApiRequest.user.details;
+
+    const receiverMatches = await getAllMatchesForUser(receiverUserDetailId);
+    const receiverUserMatches = receiverMatches.filter(
+      (match) => match._variant === MatchVariant.UserMatch
+    ) as UserMatch[];
+
+    if (receiverUserMatches.length === 0) {
+      throw new BlError("Receiver does not have any user matches");
+    }
+
+    let activeCustomerItems = [];
+    try {
+      activeCustomerItems =
+        await new CustomerItemActiveBlid().getActiveCustomerItems(blid);
+    } catch (e) {
+      throw new BlError("blid not active").code(804);
+    }
+    if (activeCustomerItems.length !== 1) {
+      throw new BlError("blid not active").code(804);
+    }
+    const customerItem = activeCustomerItems[0];
+
+    const receiverUserMatch = receiverUserMatches.find((userMatch) =>
+      userMatch.expectedItems.includes(String(customerItem.item))
+    );
+
+    if (!receiverUserMatch) {
+      throw new BlError("Item not in receiver expectedItems").code(805);
+    }
+
+    if (receiverUserMatch.receivedCustomerItems.includes(customerItem.id)) {
+      throw new BlError("Receiver has already received this item").code(806);
+    }
+
+    const senderMatches = await getAllMatchesForUser(customerItem.customer);
+    const senderUserMatches = senderMatches.filter(
+      (match) => match._variant === MatchVariant.UserMatch
+    ) as UserMatch[];
+    const senderUserMatch = senderUserMatches.find(
+      (userMatch) =>
+        userMatch.expectedItems.includes(String(customerItem.item)) &&
+        !userMatch.deliveredCustomerItems.includes(String(customerItem.item))
+    );
+
+    if (
+      senderUserMatch === undefined ||
+      receiverUserMatch.id !== senderUserMatch.id
+    ) {
+      userFeedback = `Boken du har scannet tilhørte opprinnelig en annen kunde. Boken er nå registrert på deg, men avsender må fortsatt levere sin opprinnelig bok. Ta kontakt med stand for spørsmål.`;
```

Comment:
```suggestion
      userFeedback = `Boken du har scannet tilhørte noen andre enn den du skulle motta fra. Det kan bety at hen har byttet bok med noen andre. Boken er nå registrert på deg, men hen er fortsatt ansvarlig for at den opprinnelige boken blir levert. Ta kontakt med stand for spørsmål.`;
```
I think this should really be in bl-next, but that's a bit harder to do currently and we don't have much time, so I'll allow it.

Reply:
Oops did forget an E in "opprinnelige", hope you spotted it